### PR TITLE
Esi 2220 docsbranch

### PR DESCRIPTION
**Pull request by RamJogal**

### CODE REVIEW

File: ui/src/containers/ConfigScreen/index.tsx [hunk 1]
```diff
@@ -133,14 +137,167 @@ const ConfigScreen: React.FC = function () {
...
+      normalInvalidKeys = checkMultiConfigKeys(
+        configurationData?.multi_config_keys
+      );
+      serverNormalInvalidKeys = checkMultiConfigKeys(
+        serverConfiguration?.multi_config_keys
+      );
+    }
+
+    const normalInvalidKeysList = [
+      ...Object.entries(normalInvalidKeys)
```
Comment (NamrataSane) [inline ui/src/containers/ConfigScreen/index.tsx:196]: ...Object.entries(normalInvalidKeys)
        ?.filter(([keys]) => keys?.length)
        ?.map(([source, keys]) => ({ source, keys })),
      ...Object.entries(serverNormalInvalidKeys)
        ?.filter(([keys]) => keys?.length)
        ?.map(([source, keys]) => ({ source, keys })),

File: ui/src/containers/ConfigScreen/index.tsx [hunk 2]
```diff
@@ -133,14 +137,167 @@ const ConfigScreen: React.FC = function () {
...
+        .map(([source, keys]) => ({ source, keys })),
+    ];
+
+    let otherInvalidKeysList: { source: string; keys: any[] }[] = [];
+    if (!validateOtherKeysByApi) {
+      const otherInvalidKeys = checkOtherKeys(configurationData);
+      const serverOtherInvalidKeys = checkOtherKeys(serverConfiguration);
+
+      otherInvalidKeysList = [
+        ...otherInvalidKeys.map((key) => ({
```
Comment (NamrataSane) [inline ui/src/containers/ConfigScreen/index.tsx:210]: otherInvalidKeys?.

File: ui/src/containers/ConfigScreen/index.tsx [hunk 3]
```diff
@@ -133,14 +137,167 @@ const ConfigScreen: React.FC = function () {
...
+          source: "configurationData",
+          keys: [key],
+        })),
+        ...serverOtherInvalidKeys.map((key) => ({
```
Comment (NamrataSane) [inline ui/src/containers/ConfigScreen/index.tsx:214]: serverOtherInvalidKeys?.

File: ui/src/containers/ConfigScreen/index.tsx [hunk 4]
```diff
@@ -133,14 +137,167 @@ const ConfigScreen: React.FC = function () {
...
-        localeTexts
-      );
+      const validateConfig = async () => {
+        const configScreen = rootConfig.configureConfigScreen();
+        const { multiConfigFields } = extractFieldsByConfigType(configScreen);
+
+        const { isValid, invalidKeys } = await checkValidity(
+          state?.installationData?.configuration,
+          state?.installationData?.serverConfiguration,
+          multiConfigTrueAndApiValidationEnabled.length > 0,
```
Comment (NamrataSane) [inline ui/src/containers/ConfigScreen/index.tsx:261]: multiConfigTrueAndApiValidationEnabled?.

File: ui/src/containers/ConfigScreen/index.tsx [hunk 5]
```diff
@@ -133,14 +137,167 @@ const ConfigScreen: React.FC = function () {
...
+          multiConfigFalseAndApiValidationEnabled.length > 0
```
Comment (NamrataSane) [inline ui/src/containers/ConfigScreen/index.tsx:262]: multiConfigFalseAndApiValidationEnabled?.

File: ui/src/containers/ConfigScreen/index.tsx [hunk 6]
```diff
@@ -133,14 +137,167 @@ const ConfigScreen: React.FC = function () {
...
+        );
+
+        const isDefaultKeyExist = multiConfigFields?.length
+          ? checkIsDefaultInitial(state?.installationData?.configuration)
+          : false;
+        const isMultiConfigKeysEmpty = multiConfigFields?.length
+          ? Object.keys(
+              state?.installationData?.configuration?.multi_config_keys
+            ).length === 0
```
Comment (NamrataSane) [inline ui/src/containers/ConfigScreen/index.tsx:271]: ?.length

**Review by NamrataSane — CHANGES_REQUESTED** [review]
resolve given comment

File: ui/src/containers/ConfigScreen/index.tsx (hunk 1 above)
Comment (RamJogal) [reply ui/src/containers/ConfigScreen/index.tsx:196]: Unsafe usage of optional chaining. If it short-circuits with 'undefined' the evaluation will throw TypeError

File: ui/src/containers/ConfigScreen/index.tsx (hunk 3 above)
Comment (RamJogal) [reply ui/src/containers/ConfigScreen/index.tsx:214]: Unsafe usage of optional chaining. If it short-circuits with 'undefined' the evaluation will throw TypeError

File: ui/src/containers/ConfigScreen/index.tsx (hunk 2 above)
Comment (RamJogal) [reply ui/src/containers/ConfigScreen/index.tsx:210]: Unsafe usage of optional chaining. If it short-circuits with 'undefined' the evaluation will throw TypeError

**Review by NamrataSane — APPROVED** [review]

**Pull request merged**